### PR TITLE
vm: implement CONVERT opcode

### LINKED ISSUE #881: Implement ISTYPE/CONVERT opcodes
We already have ISNULL, but we also need ISTYPE and CONVERT: neo-project/neo-vm#270, neo-project/neo-vm#273.

### PR DESCRIPTION
**Pull request by fyrchik**
Closes #881 .

### CODE REVIEW

File: pkg/vm/stack.go [hunk 1]
```diff
@@ -83,35 +83,16 @@ func (e *Element) BigInt() *big.Int {
...
-			if b != 0 {
-				return true, nil
-			}
-		}
-		return false, nil
-	case *InteropItem:
-		return t.value != nil, nil
-	default:
-		return false, fmt.Errorf("can't convert to bool: " + t.String())
+	if e.value == nil {
```
Comment (roman-khimov) [inline pkg/vm/stack.go:86]: Do we ever have valid `Element` without `value`? I don't think so.

File: pkg/vm/stack.go [hunk 2]
```diff
@@ -83,35 +83,16 @@ func (e *Element) BigInt() *big.Int {
...
+		return false, errors.New("nil value")
 	}
+	return e.value.Bool(), nil
```
Comment (roman-khimov) [inline pkg/vm/stack.go:89]: It makes `TryBool` useless. If anything can be converted to `Boolean` successfully, then we shouldn't be trying, we should just convert and be done with it.

Comment (fyrchik) [reply pkg/vm/stack.go]: Removed.

**Comment by fyrchik** [comment]
@roman-khimov I was thinking about simple type casts, but it will be rather verbose (first convert to an `interface{}`, then cast to a specific type).
I'll implement interops in this PR.

File: pkg/vm/stack.go [hunk 3]
```diff
@@ -80,38 +80,10 @@ func (e *Element) BigInt() *big.Int {
...
-		return false, nil
-	case *InteropItem:
-		return t.value != nil, nil
-	default:
-		return false, fmt.Errorf("can't convert to bool: " + t.String())
-	}
-}
-
 // Bool attempts to get the underlying value of the element as a boolean.
 // Will panic if the assertion failed which will be caught by the VM.
```
Comment (roman-khimov) [inline pkg/vm/stack.go:84]: I think we need to update this comment while we're here.

File: pkg/compiler/analysis.go [hunk 4]
```diff
@@ -17,6 +17,7 @@ var (
 		"AppCall",
 		"FromAddress", "Equals",
 		"panic",
+		"ToBool", "ToByteArray", "ToInteger",
```
Comment (roman-khimov) [inline pkg/compiler/analysis.go:20]: Am I correct that in fact plain `ToBool()` invocation would comple the same way `convert.ToBool()` does?

File: pkg/compiler/codegen.go [hunk 5]
```diff
@@ -123,7 +123,7 @@ func (c *codegen) emitLoadConst(t types.TypeAndValue) {
 
 func (c *codegen) convertBasicType(t types.TypeAndValue, typ *types.Basic) {
 	switch typ.Kind() {
-	case types.Int, types.UntypedInt, types.Uint:
+	case types.Int, types.UntypedInt, types.Uint, types.Int64:
```
Comment (roman-khimov) [inline pkg/compiler/codegen.go:126]: This is certainly not CONVERT-related and what about other types?

File: pkg/compiler/analysis.go (hunk 4 above)
Comment (fyrchik) [reply pkg/compiler/analysis.go:20]: Yes, this is a known problem with current interop implementation.

**Comment by roman-khimov** [comment]
And what do you think about `emit.Bool()`? It technically emits and integer at the moment and even though it should be casted to bool when needed, MSIL translator actually does `CONVERT` emitted bools to `Boolean`, it probably is more correct and maybe there will be some case where this difference would matter.

File: pkg/compiler/analysis.go (hunk 4 above)
Comment (roman-khimov) [reply pkg/compiler/analysis.go:20]: Do we have an issue for it?

File: pkg/compiler/codegen.go (hunk 5 above)
Comment (fyrchik) [reply pkg/compiler/codegen.go]: Fixed.

**Comment by fyrchik** [comment]
I think it is a good thing to do, but there are some circular dependencies RN (`emit` is imported into `vm` where `StackItemType` is located).

**Comment by fyrchik** [comment]
@roman-khimov check the last commit. I am not sure it belongs to `emit`, probably we can move `Stackitem` into a separate package later.

**Comment by roman-khimov** [comment]
Certainly not worth doing it this way. And `StackItem` actually uses `emit` for conversions, so we can't quickly move it away from `vm`. I think we can only safely move `StackItemType` into some separate `stacktype` package at the moment. Or just hardcode this `CONVERT` parameter and create a separate issue to untangle it.

**Pull request merged**